opentech/apply/funds/serializers.py

```diff
@@ -11,10 +12,29 @@
 class ActionSerializer(serializers.Field):
     def to_representation(self, instance):
         actions = instance.get_actions_for_user(self.context['request'].user)
-        return {
-            transition: action
-            for transition, action in actions
-        }
+        representation = []
+        for transition, action in actions:
+            action_dict = {
+                'value': transition,
+                'display': action
+            }
+
+            # Sometimes the status does not exist in the
+            # determination matrix.
+            redirect = None
+            try:
+                redirect = DeterminationCreateOrUpdateView.should_redirect(
```

Comment:
Was it not possible to extract the common method?

Reply:
Do you want me to extract should_redirect as a common method?

Reply:
Lets leave it for now. It makes sense as it is. If we make the determination view API driven then we will address that then.